modules/brainbrowser/js/brainbrowser.loris.js

```diff
@@ -748,17 +749,52 @@ $(function() {
         minc_ids_arr = [minc_ids];
     }
 
+    var request;
     for (i = 0; i < minc_ids_arr.length; i += 1) {
 
-        minc_volumes.push({
-            type: 'minc',
-	    header_url: "",
-	    raw_data_url: loris.BaseURL + "/brainbrowser/ajax/minc.php?minc_id=" + minc_ids_arr[i],
-            template: {
+      request = $.ajax({
+        dataType: "json",
+        url: loris.BaseURL + "/brainbrowser/ajax/getMincName.php",
+        data: 'minc_id=' + minc_ids_arr[i],
+        method: 'GET',
+        success: function (data) {
+          var fileid   = data.fileid;
+          var filename = data.filename;
+          if (filename == null) {
+              var msg = "ERROR: could not load the file.";
+              $("#loading").html(msg);
+              console.error(msg);
+          } else if (filename.endsWith("mnc")) {
+            minc_volumes.push({
+              type: 'minc',
+              header_url: "",
+              raw_data_url: loris.BaseURL + "/brainbrowser/ajax/minc.php?minc_id=" + fileid,
+              template: {
                 element_id: "volume-ui-template4d",
                 viewer_insert_class: "volume-viewer-display"
-            }
-        });
+              }
+            });
+          } else if (filename.endsWith("nii")) {
```

Comment:
I'd prefer this was derived from the `Content-Type` header of the response rather than the filename, but this should work.

Reply:
@driusan should be done with the latest commit